blocks/bricks/conv.py

```diff
@@ -207,16 +220,105 @@ def apply(self, input_):
             with the last two dimensions downsampled.
 
         """
-        output = max_pool_2d(input_, self.pooling_size, st=self.step)
+        output = max_pool_2d(input_, self.pooling_size, st=self.step,
```

Comment:
As for why I went the route of "base class" rather than "abstract base class" with an abstract apply method... I think the implementation makes it clear. Basically MaxPooling and AveragePooling just need different constructors that call different the superclass constructor with different arguments, and everything else remains the same.